File: vactrack/lib/models/parent.dart
import 'package:vactrack/models/child.dart';

class Parent {
  final String email;
  final String? name;
  final String? adahar;
  final String? location;
  final List<Child> children;

  Parent({
    required this.email,
    this.name,
    this.adahar,
    this.location,
    required this.children,
  });

  factory Parent.fromJson(Map<String, dynamic> data) {
    return Parent(
      email: data['email'],
      name: data['name'],
      adahar: data['adahar'],
      location: data['location'],
      children: (data['children'] as List)
          .map((childData) => Child.fromJson(childData))
          .toList(),
    );
  }


  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'name': name,
      'adahar': adahar,
      'location': location,
      'children': children.map((child) => child.toJson()).toList(),
    };
  }
}

File: backend/src/models/parent.ts
// models/Parent.ts
import { Model, Table, Column, DataType, HasMany, AllowNull, BeforeCreate } from 'sequelize-typescript';
import { Child } from './child';
import bcrypt from 'bcrypt';


@Table({
  tableName: 'parents',
  timestamps: false, // Change to true if you need createdAt/updatedAt
})
export class Parent extends Model {
  @Column({
    type: DataType.STRING,
    allowNull: true,
  })
  name?: string;

  @Column({
    type: DataType.STRING,
    unique: true,
    allowNull: false,
  })
  email!: string;

  @Column({
    type: DataType.STRING,
    allowNull: true,
  })
  adahar?: string;

  @Column({
    type: DataType.STRING,
    allowNull: true,
  })
  location?: string;

  @Column({
      type: DataType.STRING,
      allowNull: false,
  })
  password!: string; 


  @HasMany(() => Child)
  children!: Child[];

  @BeforeCreate
  static async hashPassword(parent: Parent){
      parent.password = await bcrypt.hash(parent.password,10);
  }
}
